### PR TITLE
WIP: Graphite: Explore UI fixes

### LINKED ISSUE #91176: Hitting enter on the graphite query editor casues the interface to throw away the full input and use `*`
### What happened?

Using the Graphite query editor, the segments give a drop-down display. I can use free text in the display, which filters down the list of optionals in the dropdown. The selected option is always `*` at the top.

* I type `carbon` into the box, the dropdown shows `*` and `carbon`, with `*` selected. I hit enter, and the interface removes my `carbon`, and uses `*` instead.
 * I type `carbon` into the box as above, press down and then enter. The interface correctly adds `carbon` into the segment.
* I type `carb*` into the box, the dropdown shows `*`, `carbon` and `carb* (Hit enter to add)`. I hit enter, and the interface removes my `carb*`, and uses `*` instead.
  * I type `carb*` into the box as above, press down twice and press enter. The interface correctly adds this element.
  * I type `carb*` into the box again, and the "Hit enter to add" option isn't available at all -- I'm no longer able to type in the current value of the field (pressing Escape restores it, but this isn't obvious as the old value is hidden at this point).
* I type and save a complex value into the box by manually navigating to 'Hit enter to add'. I now click the box again. My input is not pre-filled into the box. I need to start again or use the pencil mode.

output

### What did you expect to happen?

* In all cases above, the UI should use what I've typed as the final value.
* When editing the field again, the UI should remember the previous value, not start from an empty state.
* (Less important) The _previous_ value should be available to revert to in the list; it shouldn't be special cased to not be selectable if and only if it's a nontrivial wildcard.

### Did this work before?

I think this was working fine around Grafana 6 or Grafana 7. It's been broken for quite a long time.

### How do we reproduce it?

See instructions above. Just add any graphite source -- I tested this with Grafana 11.1.3 on a fresh Graphite install, but it's also reproducible on our production instance.

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 11.1.3
OS: Debian GNU/Linux 12, Ubuntu 22.04
Browser: Firefox, Chromium

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

### Datasource(s)?

Graphite

### PR DESCRIPTION
**Pull request by bucko909**
Contains a number of bugfixes/improvements to the Graphite metric selection UI.

Fixes #91176

### CODE REVIEW

**Comment by bucko909** [comment]
Please see #91176 for more info. I've fixed only some of the issues, and need advice/help on the remainder.

**Comment by bucko909** [comment]
@JoaoSilvaGrafana Sorry, just picking a random assignee as I'm a bit stuck as to why this hasn't moved. Do I need to prod this somehow to get it moved forwards?

**Comment by bucko909** [comment]
Oh, seems like the ticket moved just 2 hours ago. I guess this is doing something, then?

**Comment by JoaoSilvaGrafana** [comment]
Hey @bucko909 , sorry I completely missed this 😓 
I would say we should be skeptical of adding new props to `grafana-ui` components, we don't want to have them become too bloated and working different everywhere, for example after this change, some `Segment` will have the create option at the top and others at the bottom.
@tskarhed is working on building a new Select component, I am not sure if we have discussed having a create option or not and if that would show up at the top or bottom. This should be consistent across Grafana and if we want it at the top it should be at the top for every select. Would be interesting to get his input here 😄

**Comment by bucko909** [comment]
Note I'm not too bothered about the top/bottom-ness of the custom option, just that it should be the default to make the UI easier to use. It especially should be the default when it comes with a tip of "Hit enter to add"!

File: packages/grafana-ui/src/components/Select/SelectBase.tsx [hunk 1]
```diff
@@ -358,16 +358,24 @@ export function SelectBase<T, Rest = {}>({
...
-      <div style={{ flexGrow: 1 }} />
-      <div className="muted small" style={{ display: 'flex', gap: '8px', alignItems: 'center' }}>
-        Hit enter to add
+function defaultFormatCreateLabel(createOptionPosition: 'first' | 'last') {
+  let hint;
+  if (createOptionPosition === 'first') {
+    hint = 'Hit enter to add';
+  } else {
+    hint = 'Click to add';
+  }
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Select/SelectBase.tsx:367]: This is a great change. It doesn't even need to be dependant on `createOptionPosition` - the message should just be changed to "Select to add"

**Review by joshhunt — CHANGES_REQUESTED** [review]
Hi @bucko909 - thanks for the contribution.

This PR has been tricky for us to review because it loaded a few too many changes into one PR, so I'm going to close it for now.

If you're still keen to get some of these changes in, I would recommended submitting seperate PRs for each 'change'. Importantly, changes to Graphite logic must be seperate from `@grafana/ui` changes so we can get the two through independently.

I hope I don't discourage you from making future contributions, but I think this is the least fuss way to get changes though 👍

**Pull request closed without merging**